src/InputFileResolver.ts

```diff
@@ -36,12 +37,12 @@ export default class InputFileResolver {
     console.log(`WARNING: Globbing expression "${expression}" did not result in any files.`)
   }
 
-  private static resolveFileGlobs(sourceExpressions: string[], resultFiles: string[]): Promise<void[]> {
-    return Promise.all(sourceExpressions.map((mutateFileExpression: string) => glob(mutateFileExpression).then(files => {
+  private static resolveFileGlobs(sourceExpressions: string[]): Promise<string[]> {
+    return Promise.all(sourceExpressions.map(expression => glob(expression).then(files => {
       if (files.length === 0) {
-        this.reportEmptyGlobbingExpression(mutateFileExpression);
+        this.reportEmptyGlobbingExpression(expression);
       }
-      files.forEach(f => resultFiles.push(f));
```

Comment:
This was the old way of pusing files to the result array. Per `expression` a globbing promise get's resolved. The promisses get created in the correct order, but the resolving can happen in any order. The pushing on the `resultFiles` array will in a random order.